buildtest/tools/build.py

```diff
@@ -141,28 +155,29 @@ def run_tests(build_id):
     runfile = datetime.now().strftime("buildtest_%H_%M_%d_%m_%Y.run")
     run_output_file = os.path.join(test_dir, "run", runfile)
     create_dir(os.path.join(test_dir, "run"))
-    fd = open(run_output_file, "w")
+
     count_test = len(tests)
     passed_test = 0
     failed_test = 0
 
-    for test in tests:
-        ret = subprocess.Popen(
-            test, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
-        )
-        output = ret.communicate()[0].decode("utf-8")
+    with open(run_output_file, "w") as fd:
+        for test in tests:
+            ret = subprocess.Popen(
```

Comment:
``subprocess.Popen`` works for now, eventually once we fix #182 we should refactor all code. Looks good with me for now